extensions/amp-list/0.1/amp-list.js

```diff
@@ -618,7 +618,22 @@ export class AmpList extends AMP.BaseElement {
       .then(
         response => {
           userAssert(
-            response && typeof response['html'] === 'string',
+            response,
+            'Error proxying amp-list templates, received no response.'
+          );
+          const init = response['init'];
+          if (init) {
+            const status = init['status'];
+            if (status >= 300) {
+              /** HTTP status codes of 300+ mean redirects and errors. */
+              throw user().createError(
+                'Error proxying amp-list templates with status: ',
+                status
+              );
+            }
+          }
+          userAssert(
+            typeof response['html'] === 'string',
             'Expected response with format {html: <string>}. Received: ',
```

Comment:
Should this error message be more developer-friendly?

When I use `amp-list` without specifying a `<tempate>`, I get this error instead in the console:
```
error.js:195 Template not found for  <amp-list src=​"https:​/​/​amp.dev/​static/​samples/​json/​cart.json" layout=​"fixed-height" height=​"80" class=​"i-amphtml-element i-amphtml-layout-fixed-height i-amphtml-layout-size-defined i-amphtml-layout i-amphtml-error i-amphtml-element-error" i-amphtml-layout=​"fixed-height" aria-live=​"polite" style=​"height:​ 80px;​ --loader-delay-offset:​8ms !important;​" error-message=​"Template not found for amp-list&#8203;​&#8203;​&#8203;​">​…​</amp-list>​
rf @ error.js:195
```

Reply:
Filed #25360 for now